<robot name="vs050">

  <link name="R_base_link">
    <!-- <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="6.0925" />
      <inertia ixx="1" ixy="-0.00" ixz="0.00" iyy="1" iyz="0" izz="1" />
    </inertial> -->
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://assisted_suturing_control/urdf/meshes_r/base_link.sldprt.STL" />
      </geometry>
      <material name="">
        <color rgba="1.0 1.0 1.0 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://assisted_suturing_control/urdf/meshes_r/base_link.sldprt.STL" />
      </geometry>
    </collision>
  </link>

  <!--  -->
  <link name="R_link1">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="2.8351" />
      <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1" />
    </inertial>
    <visual>
      <origin xyz="-0.035 0 0.345" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://assisted_suturing_control/urdf/meshes_r/link1.sldprt.STL" />
      </geometry>
      <material name="">
        <color rgba="1.0 1.0 1.0 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="-0.035 0 0.345" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://assisted_suturing_control/urdf/meshes_r/link1.sldprt.STL" />
      </geometry>
    </collision>
  </link>

  <joint name="R_joint_1" type="revolute">
    <parent link="R_base_link"/>
    <child link="R_link1"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0" />
    <axis xyz="0.0 0.0 1.0" />
    <limit effort="1" lower="-1.570796" upper="1.570796" velocity="1" />
    <dynamics damping="0" friction="0" />
  </joint>

  <!--  -->
  <link name="R_link2">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="3.4626" />
      <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://assisted_suturing_control/urdf/meshes_r/link2.sldprt.STL" />
      </geometry>
      <material name="">
        <color rgba="1.0 1.0 1.0 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://assisted_suturing_control/urdf/meshes_r/link2.sldprt.STL" />
      </geometry>
    </collision>
  </link>

  <joint name="R_joint_2" type="revolute">
    <parent link="R_link1"/>
    <child link="R_link2"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.345" />
    <axis xyz="0.0 1.0 0.0" />
    <limit effort="1" lower="-0.087266" upper="1.570796" velocity="1" />
    <dynamics damping="0" friction="0" />
  </joint>

  <!--  -->
  <link name="R_link3">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="1.492" />
      <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://assisted_suturing_control/urdf/meshes_r/link3.sldprt.STL" />
      </geometry>
      <material name="">
        <color rgba="1.0 1.0 1.0 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://assisted_suturing_control/urdf/meshes_r/link3.sldprt.STL" />
      </geometry>
    </collision>
  </link>

  <joint name="R_joint_3" type="revolute">
    <parent link="R_link2"/>
    <child link="R_link3"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.25" />
    <axis xyz="0.0 1.0 0.0" />
    <limit effort="1" lower="0" upper="2.1642" velocity="1" />
    <dynamics damping="0" friction="0" />
  </joint>

  <!--  -->
  <link name="R_link4">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="0.98124" />
      <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1" />
    </inertial>
    <visual>
      <origin xyz="0 0 0.850" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://assisted_suturing_control/urdf/meshes_r/link4.sldprt.STL" />
      </geometry>
      <material name="">
        <color rgba="1.0 1.0 1.0 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0.850" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://assisted_suturing_control/urdf/meshes_r/link4.sldprt.STL" />
      </geometry>
    </collision>
  </link>

  <joint name="R_joint_4" type="revolute">
    <parent link="R_link3"/>
    <child link="R_link4"/>
    <origin rpy="0.0 0.0 0.0" xyz="-0.01 0.0 -0.595" />
    <axis xyz="0.0 0.0 1.0" />
    <limit effort="1" lower="0" upper="3.1416" velocity="1" />
    <dynamics damping="0" friction="0" />
  </joint>

  <!--  -->
  <link name="R_link5">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="0.41203" />
      <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://assisted_suturing_control/urdf/meshes_r/link5.sldprt.STL" />
      </geometry>
      <material name="">
        <color rgba="0.4 0.4 0.4 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://assisted_suturing_control/urdf/meshes_r/link5.sldprt.STL" />
      </geometry>
    </collision>
  </link>

  <joint name="R_joint_5" type="revolute">
    <parent link="R_link4"/>
    <child link="R_link5"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.85" />
    <axis xyz="0.0 1.0 0.0" />
    <limit effort="1" lower="-1.570796" upper="1.570796" velocity="1" />
    <dynamics damping="0" friction="0" />
  </joint>

  <!--  -->
  <link name="R_4dof_base">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="0" />
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://assisted_suturing_control/urdf/meshes_r/forceps_base_link_right_v2.STL" />
      </geometry>
      <material name="">
        <color rgba="0.7 0.7 0.7 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://assisted_suturing_control/urdf/meshes_r/forceps_base_link_right_v2.STL" />
      </geometry>
    </collision>
  </link>

  <joint name="R_joint_6" type="revolute">
    <parent link="R_link5"/>
    <child link="R_4dof_base"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.07" />
    <axis xyz="0.0 0.0 1.0" />
    <limit effort="1" lower="-1.570796" upper="1.570796" velocity="1" />
    <dynamics damping="0" friction="0" />
  </joint>

  <!--  -->
  <link name="R_4dof_j1">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="0.0014724" />
      <inertia ixx="5.82003123944285E-06" ixy="3.24149380466014E-12" ixz="1.83969021647371E-14" iyy="5.82004751970887E-06" iyz="-2.62549260457254E-14" izz="1.66321589719335E-09" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0.0 0" />
      <geometry>
        <mesh filename="package://assisted_suturing_control/urdf/meshes_r/4dof_j1.STL" />
      </geometry>
      <material name="">
        <color rgba="1.0 1.0 0.70 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0.0 0" />
      <geometry>
        <mesh filename="package://assisted_suturing_control/urdf/meshes_r/4dof_j1.STL" />
      </geometry>
    </collision>
  </link>

  <joint name="R_joint_7" type="revolute">
    <parent link="R_4dof_base"/>
    <child link="R_4dof_j1"/>
    <origin rpy="0.0 -0.7853981633 0.0" xyz="0.05066 -0.00010 0.1341" />
    <axis xyz="0.0 0.0 1.0" />
    <limit effort="1" lower="-3.1416" upper="3.1416" velocity="1" />
    <dynamics damping="0" friction="0" />
  </joint>

  <!--  -->
  <link name="R_4dof_j2">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="1.0254E-05" />
      <inertia ixx="1.46123926133104E-11" ixy="5.39313158039792E-17" ixz="-5.89789608147636E-16" iyy="1.77686557711243E-11" iyz="2.30207294092127E-16" izz="1.37632621709668E-11" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://assisted_suturing_control/urdf/meshes_r/4dof_j2.STL" />
      </geometry>
      <material name="">
        <color rgba="1.0 1.0 0.70 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://assisted_suturing_control/urdf/meshes_r/4dof_j2.STL" />
      </geometry>
    </collision>
  </link>

  <joint name="R_joint_8" type="revolute">
    <parent link="R_4dof_j1"/>
    <child link="R_4dof_j2"/>
    <!-- <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.21704999" />
    <axis xyz="0.0 1.0 0.0" /> -->
    <origin rpy="-1.570796327 0.0 0.0" xyz="0.0 0.0 0.21704999" />
    <axis xyz="0.0 0.0 1.0" />
    <limit effort="1" lower="-0.785398" upper="0.785398" velocity="1" />
    <!-- <limit effort="1" lower="-0.785398" upper="0.785398" velocity="1" /> -->
    <dynamics damping="0" friction="0" />
  </joint>

  <!--  -->
  <link name="R_4dof_j3">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="4.3315E-06" />
      <inertia ixx="1.44635859506575E-11" ixy="-1.7433432504869E-13" ixz="1.2873000331331E-12" iyy="1.40311701607746E-11" iyz="2.00519167620474E-12" izz="1.7319134927659E-12" />
    </inertial>
    <visual name="gripper1">
      <origin xyz="0 0 0" rpy="0 0.0 0" />
      <geometry>
        <mesh filename="package://assisted_suturing_control/urdf/meshes_r/4dof_j3.STL" />
      </geometry>
      <material name="">
        <color rgba="1.0 0.2 0.2 1" />
      </material>
    </visual>
    <visual name="gripper2">
      <origin xyz="0 0 0" rpy="0 0.0 0" />
      <geometry>
        <mesh filename="package://assisted_suturing_control/urdf/meshes_r/4dof_j4.STL" />
      </geometry>
      <material name="">
        <color rgba="0.2 1.0 0.2 1" />
      </material>
    </visual>
    <collision name="gipper1_col">
      <origin xyz="0 0 0" rpy="0 0.0 0" />
      <geometry>
        <mesh filename="package://assisted_suturing_control/urdf/meshes_r/4dof_j3.STL" />
      </geometry>
    </collision>
    <collision name ="gipper2_col">
      <origin xyz="0 0 0" rpy="0 0.0 0" />
      <geometry>
        <mesh filename="package://assisted_suturing_control/urdf/meshes_r/4dof_j4.STL" />
      </geometry>
    </collision>
  </link>

  <joint name="R_joint_9" type="revolute">
    <parent link="R_4dof_j2"/>
    <child link="R_4dof_j3"/>
    <origin rpy="1.570796327 0.0 1.570796327" xyz="0.0 -0.0023 0.0" />
    <axis xyz="0.0 0.0 1.0" />
    <limit effort="1" lower="-0.785398" upper="0.785398" velocity="1" />
    <!-- <limit effort="1" lower="-0.785398" upper="0.785398" velocity="1" /> -->
    <dynamics damping="0" friction="0" />
  </joint>

  <!--  -->
  <link name="R_ee">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="4.3315E-06" />
      <inertia ixx="1.44635859506575E-11" ixy="-1.7433432504869E-13" ixz="1.2873000331331E-12" iyy="1.40311701607746E-11" iyz="2.00519167620474E-12" izz="1.7319134927659E-12" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0.0 0" />
      <geometry>
        <sphere radius="0.0005" />
        <material>
          <color rgba="0.2 1.0 0.2 1" />
        </material>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0.0 0" />
      <geometry>
        <sphere radius="0.0005" />
      </geometry>
    </collision>
  </link>

  <joint name="R_joint_10" type="fixed">
    <parent link="R_4dof_j3"/>
    <child link="R_ee"/>
    <origin rpy="0.0 0.0 0.0" xyz="-0.0035 0.0 0.0" />
  </joint>

  <!--  -->
  <link name="R_Fj7">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="4.3315E-06" />
      <inertia ixx="1.44635859506575E-11" ixy="-1.7433432504869E-13" ixz="1.2873000331331E-12" iyy="1.40311701607746E-11" iyz="2.00519167620474E-12" izz="1.7319134927659E-12" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0.0 0" />
      <geometry>
        <sphere radius="0.0005" />
        <material>
          <color rgba="0.2 1.0 0.2 1" />
        </material>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0.0 0" />
      <geometry>
        <sphere radius="0.0005" />
      </geometry>
    </collision>
  </link>

  <joint name="R_joint_11" type="fixed">
    <parent link="R_4dof_base"/>
    <child link="R_Fj7"/>
    <origin rpy="0.0 -0.7853981633 0.0" xyz="0.05066 -0.00010 0.1341" />
  </joint>

  <!--  -->
  <link name="R_Fj8">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="4.3315E-06" />
      <inertia ixx="1.44635859506575E-11" ixy="-1.7433432504869E-13" ixz="1.2873000331331E-12" iyy="1.40311701607746E-11" iyz="2.00519167620474E-12" izz="1.7319134927659E-12" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0.0 0" />
      <geometry>
        <sphere radius="0.0005" />
        <material>
          <color rgba="0.2 1.0 0.2 1" />
        </material>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0.0 0" />
      <geometry>
        <sphere radius="0.0005" />
      </geometry>
    </collision>
  </link>

  <joint name="R_joint_12" type="fixed">
    <parent link="R_4dof_j1"/>
    <child link="R_Fj8"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.21704999" />
  </joint>

</robot>